### DISCUSSION TITLE
Default type for generic classes

### DISCUSSION THREAD

**Discussion by tidyui** [dotnet/csharplang#6118] [Language Ideas] [answered]
For simplicity and readability it would be nice to be able to specify a default type for a generic class, for example:

~~~ csharp
public class MyClass<T = MyDefaultClass>
{
   ...
}
~~~

Doing this would expose the class to the compiler both as `MyClass<T>` and `MyClass` and it would be possible to do the following:

~~~ csharp
var instance1 = new MyClass<MyItemClass>();
var instance2 = new MyClass();
~~~

With this not available I very often find it necessary to add the following code in the projects.

~~~ csharp
public class MyClass<T> where T : MyBaseClass
{
   ...
}

public MyClass : MyClass<MyDefaultClass>
{
   ...
}

public abstract class MyBaseClass
{
   ...
}

public class MyDefaultClass : MyBaseClass
{
   ...
}
~~~

Not only is it unnecessary, but it's also hard to make it visible from the project as both of these classes (`MyClass<T>` & `MyClass`) most often ends up in the same file (`MyClass.cs`).

**Comment by HaloFour** [comment, marked as answer]
See: #278
